start.sh

```diff
@@ -13,17 +15,36 @@ echo "Push to branch $INPUT_BRANCH";
     exit 1;
 };
 
+if ${INPUT_FORCE} && ${INPUT_FORCE_WITH_LEASE}; then
+  echo 'Please, specify only force or force_with_lease and not both.';
+  exit 1;
+fi
+
 if ${INPUT_FORCE}; then
     _FORCE_OPTION='--force'
 fi
 
+if ${INPUT_FORCE_WITH_LEASE}; then
+    _FORCE_OPTION='--force-with-lease'
+fi
+
 if ${INPUT_TAGS}; then
     _TAGS='--tags'
 fi
 
 cd ${INPUT_DIRECTORY}
 
-remote_repo="${INPUT_GITHUB_URL_PROTOCOL}//${GITHUB_ACTOR}:${INPUT_GITHUB_TOKEN}@${INPUT_GITHUB_URL}/${REPOSITORY}.git"
+if ${INPUT_SSH}; then
+    remote_repo="git@${INPUT_GITHUB_URL}:${REPOSITORY}.git"
+else
+    remote_repo="${INPUT_GITHUB_URL_PROTOCOL}//${GITHUB_ACTOR}:${INPUT_GITHUB_TOKEN}@${INPUT_GITHUB_URL}/${REPOSITORY}.git"
+fi
+
 git config --local --add safe.directory ${INPUT_DIRECTORY}
 
-git push "${remote_repo}" HEAD:${INPUT_BRANCH} --follow-tags $_FORCE_OPTION $_TAGS;
+#if ${INPUT_FORCE_WITH_LEASE}; then
```

Comment:
Do we need the commented lines

Reply:
I'll uncomment the lines/ revert the tested modifications and squash the changes together.

Reply:
Okay. Perfect